storage/perfschema/pfs_setup_actor.cc

```diff
@@ -175,15 +177,15 @@ int insert_setup_actor(const String *user, const String *host, const String *rol
   if (unlikely(pins == NULL))
     return HA_ERR_OUT_OF_MEM;
 
-  static uint PFS_ALIGNED setup_actor_monotonic_index= 0;
+  static std::atomic<uint> setup_actor_monotonic_index(0);
```

Comment:
please return PFS_ALIGNED back. It aligns the variable to the L1 cache line size, and this variable is static, and besides inside the function, so its layout is not clear (to me at least), and it can be false-shared with some other global variables.